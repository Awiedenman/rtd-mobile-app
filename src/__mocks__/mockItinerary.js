export const mockItinerary =[{
  itinerary_id: 7,
  start_address: "Denver, CO 80202, USA",
  end_address: "366 Broadway, Denver, CO 80203, USA",
  favorite: false,
  departure_time: "3:51pm",
  arrival_time: "4:11pm",
  duration: "20 mins",
  distance: "3.0 mi",
  steps: [{
    id: 26,
    possible_route_id: 7,
    headsign: null,
    arrival_time: null,
    departure_time: null,
    arrival_stop: null,
    departure_stop: null,
    name: null,
    short_name: null,
    instructions: "Walk to Union Sta,tion",
    distance: "1 ft",
    duration: "1 min",
    color: null,
    num_stops: null,
    vehicle_type: null,
    credit_name: null,
    credit_url: null,
    travel_mode: "Walking"
  },
  {
    id: 27,
    possible_route_id: 7,
    headsign: "Highlands Ranch",
    arrival_time: "4:11pm",
    departure_time: "3:52pm",
    arrival_stop: "Broadway & 4th Ave",
    departure_stop: "Union Station",
    name: "South Broadway",
    short_name: "0",
    instructions: "Bus towards Highlands Ranch",
    distance: "3.0 mi",
    duration: "20 mins",
    color: "#0277bd",
    num_stops: 11,
    vehicle_type: "Bus",
    credit_name: "Regional Transportation District",
    credit_url: "http://rtd-denver.com/",
    travel_mode: "TRANSIT"
},
{
    id: 28,
    possible_route_id: 7,
    headsign: null,
    arrival_time: null,
    departure_time: null,
    arrival_stop: null,
    departure_stop: null,
    name: null,
    short_name: null,
    instructions: "Walk to 366 Broadway, Denver, CO 80203, USA",
    distance: "108 ft",
    duration: "1 min",
    color: null,
    num_stops: null,
    vehicle_type: null,
    credit_name: null,
    credit_url: null,
    travel_mode: "Walking"
  }]

}]