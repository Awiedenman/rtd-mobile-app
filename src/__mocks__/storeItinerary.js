export const storeItinerary = jest.fn();