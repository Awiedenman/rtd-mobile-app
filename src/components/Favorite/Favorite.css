.favorite_card {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(231, 231, 231, 0.9);
  width: 95%;
  height: 40px;
  margin: 1rem auto;
}

h2 {
  font-size: 130%;
  color: #213982;
}

.favorite_trip-details {
  
} 

.starting-point {

} 

.destination {

}

