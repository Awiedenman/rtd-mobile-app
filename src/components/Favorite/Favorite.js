import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Favorite.css'; 

export class Favorite extends Component {

  render() {
    const { 
      favData, 
      searchFavorite,
      deleteFavorite
    } = this.props;

    const {
      start_address,
      end_address,
      id
    } = favData;

    return (
      <div className='favorite-button-container'>
        <button
          className='favorite-button'
          value={id}
          onClick={searchFavorite}
        >
          <div className='address-container'>
            <p className='from-to'>From:</p>
            <p className='address'>
              {start_address.split(',')[0]}
            </p>
          </div>
          <div className='address-container'>
            <p className='from-to'>To:</p>
            <p className='address'>
              {end_address.split(',')[0]}
            </p>
          </div>
        </button>
        <button 
          className='favorite-delete-button'
          onClick={deleteFavorite}
          id={id}
        >
        </button>
      </div>
    );
  }
}

export default Favorite;

Favorite.propTypes = {
  name: PropTypes.string,
  start_address: PropTypes.string,
  end_address: PropTypes.string,
  itinerary_id: PropTypes.string,
  isLoading: PropTypes.string,
  searchFavorite: PropTypes.func,
  deleteFavorite: PropTypes.func
}; 