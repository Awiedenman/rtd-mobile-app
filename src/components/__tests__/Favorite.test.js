import React from 'react';
import { shallow } from 'enzyme';
import { Favorite } from '../Favorite/Favorite';
import { mockItinerary } from '../../__mocks__/mockItinerary';

describe('Favorite', () => {
  it('should match snapShot', () => {
    const mockName = 'austin';
    const mockData = mockItinerary;
    const mockStart = 'union station, denver, co';
    // const mockEnd = 'overland golf course';
    const wrapper = shallow(
      <Favorite 
        name={mockName}
        favData={[mockData]}
        startAddress ={mockStart}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
