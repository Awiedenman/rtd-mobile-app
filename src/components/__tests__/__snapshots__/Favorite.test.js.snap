// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorite
    deleteFavorite={[Function]}
    favData={
        Object {
            "end_address": "1700 Wynkoop St, Denver, CO 80202, USA",
            "favorite": true,
            "id": 88,
            "start_address": "2201, 1100 14th St, Denver, CO 80202, United States",
          }
    }
    searchFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorite-button"
          onClick={[Function]}
          value={88}
>
          <div
                    className="address-container"
          >
                    <p
                              className="from-to"
                    >
                              From:
                    </p>
                    <p
                              className="address"
                    >
                              2201
                    </p>
          </div>
          <div
                    className="address-container"
          >
                    <p
                              className="from-to"
                    >
                              To:
                    </p>
                    <p
                              className="address"
                    >
                              1700 Wynkoop St
                    </p>
          </div>
</button>,
        <button
          className="favorite-delete-button"
          id={88}
          onClick={[Function]}
/>,
      ],
      "className": "favorite-button-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="address-container"
>
              <p
                            className="from-to"
              >
                            From:
              </p>
              <p
                            className="address"
              >
                            2201
              </p>
</div>,
            <div
              className="address-container"
>
              <p
                            className="from-to"
              >
                            To:
              </p>
              <p
                            className="address"
              >
                            1700 Wynkoop St
              </p>
</div>,
          ],
          "className": "favorite-button",
          "onClick": [Function],
          "value": 88,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="from-to"
>
                  From:
</p>,
                <p
                  className="address"
>
                  2201
</p>,
              ],
              "className": "address-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "From:",
                  "className": "from-to",
                },
                "ref": null,
                "rendered": "From:",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2201",
                  "className": "address",
                },
                "ref": null,
                "rendered": "2201",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="from-to"
>
                  To:
</p>,
                <p
                  className="address"
>
                  1700 Wynkoop St
</p>,
              ],
              "className": "address-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "To:",
                  "className": "from-to",
                },
                "ref": null,
                "rendered": "To:",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1700 Wynkoop St",
                  "className": "address",
                },
                "ref": null,
                "rendered": "1700 Wynkoop St",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "favorite-delete-button",
          "id": 88,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorite-button"
            onClick={[Function]}
            value={88}
>
            <div
                        className="address-container"
            >
                        <p
                                    className="from-to"
                        >
                                    From:
                        </p>
                        <p
                                    className="address"
                        >
                                    2201
                        </p>
            </div>
            <div
                        className="address-container"
            >
                        <p
                                    className="from-to"
                        >
                                    To:
                        </p>
                        <p
                                    className="address"
                        >
                                    1700 Wynkoop St
                        </p>
            </div>
</button>,
          <button
            className="favorite-delete-button"
            id={88}
            onClick={[Function]}
/>,
        ],
        "className": "favorite-button-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="address-container"
>
                <p
                                className="from-to"
                >
                                From:
                </p>
                <p
                                className="address"
                >
                                2201
                </p>
</div>,
              <div
                className="address-container"
>
                <p
                                className="from-to"
                >
                                To:
                </p>
                <p
                                className="address"
                >
                                1700 Wynkoop St
                </p>
</div>,
            ],
            "className": "favorite-button",
            "onClick": [Function],
            "value": 88,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="from-to"
>
                    From:
</p>,
                  <p
                    className="address"
>
                    2201
</p>,
                ],
                "className": "address-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "From:",
                    "className": "from-to",
                  },
                  "ref": null,
                  "rendered": "From:",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "2201",
                    "className": "address",
                  },
                  "ref": null,
                  "rendered": "2201",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="from-to"
>
                    To:
</p>,
                  <p
                    className="address"
>
                    1700 Wynkoop St
</p>,
                ],
                "className": "address-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "To:",
                    "className": "from-to",
                  },
                  "ref": null,
                  "rendered": "To:",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1700 Wynkoop St",
                    "className": "address",
                  },
                  "ref": null,
                  "rendered": "1700 Wynkoop St",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "favorite-delete-button",
            "id": 88,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
